popcorn/apps/planner/__init__.py

```diff
@@ -52,114 +42,106 @@ def setup_defaults(self, loglevel=None, logfile=None, queue=None, strategy=None,
 
 
 class Planner(object):
+    instances = {
+
+    }  # key queue name , value : instance
+
+    threading_pool = {
 
-    class Blueprint(bootsteps.Blueprint):
+    }  # key: planner instance, value : planner's thread
 
-        """Hub bootstep blueprint."""
-        name = 'Planner'
-        default_steps = set([
-            'popcorn.rpc.pyro:RPCClient',
-            'popcorn.apps.planner:Strategy',
-            'popcorn.apps.planner:Loop',
-        ])
+    @classmethod
+    def get_instance(cls, app, queue, strategy_name):
+        ins = cls.instances.get(queue)
+        if not ins:
+            ins = Planner(app, queue, strategy_name)
+            cls.instances[queue] = ins
+            info('[Planner] - [Duplicated] %s', queue)
```

Comment:
This will be triggered every time when register new planner. Should be a bug, we call get_instance() twice each time.